src/core/public/http/types.ts

```diff
@@ -18,84 +18,230 @@
   keepalive?: boolean;
+
+  /** HTTP method, which is "GET" by default. */
   method?: string;
+
+  /**
+   * The mode associated with request, which is a string indicating whether the request will use CORS, or will be
+   * restricted to same-origin URLs.
+   */
   mode?: RequestMode;
+
+  /**
+   * The redirect mode associated with request, which is a string indicating how redirects for the request will be
+   * handled during fetching. A request will follow redirects by default.
+   */
   redirect?: RequestRedirect;
+
+  /**
+   * The referrer of request. Its value can be a same-origin URL if explicitly set in init, the empty string to
+   * indicate no referrer, and "about:client" when defaulting to the global's default. This is used during fetching to
+   * determine the value of the `Referer` header of the request being made.
+   */
   referrer?: string;
+
+  /**
+   * The referrer policy associated with request. This is used during fetching to compute the value of the request's
+   * referrer.
+   */
   referrerPolicy?: ReferrerPolicy;
+
+  /**
+   * Returns the signal associated with request, which is an AbortSignal object indicating whether or not request has
+   * been aborted, and its abort event handler.
+   */
   signal?: AbortSignal | null;
-  window?: any;
-}
-/** @public */
-export interface HttpDeps {
-  injectedMetadata: InjectedMetadataSetup;
-  fatalErrors: FatalErrorsSetup | null;
+
+  /**
+   * Can only be null. Used to disassociate request from any Window.
+   */
+  window?: null;
 }
+
 /** @public */
 export interface HttpFetchQuery {
   [key: string]: string | number | boolean | undefined;
 }
-/** @public */
+
+/**
+ * All options that may be used with a {@link HttpHandler}.
+ * @public
+ */
 export interface HttpFetchOptions extends HttpRequestInit {
+  /**
+   * The query string for an HTTP request. See {@link HttpFetchQuery}.
+   */
   query?: HttpFetchQuery;
+
+  /**
+   * Whether or not the request should automatically prepend the basePath. Defaults to `true`.
+   */
   prependBasePath?: boolean;
+
+  /**
+   * Headers to send with the request. See {@link HttpHeadersInit}.
+   */
   headers?: HttpHeadersInit;
 }
-/** @public */
+
+/**
+ * A function for making an HTTP requests to Kibana's backend. See {@link HttpFetchOptions} for options and
+ * {@link HttpBody} for the response.
+ *
+ * @param path the path on the Kibana server to send the request to. Should not include the basePath.
+ * @param options {@link HttpFetchOptions}
+ * @returns a Promise that resolves to a {@link HttpBody}
+ * @public
+ */
 export type HttpHandler = (path: string, options?: HttpFetchOptions) => Promise<HttpBody>;
+
 /** @public */
 export type HttpBody = BodyInit | null | any;
+
 /** @public */
 export interface HttpResponse {
   request?: Request;
   response?: Response;
   body?: HttpBody;
 }
+
+/** @public */
+export interface IHttpFetchError extends Error {
```

Comment:
APM wanted to type their fetch exceptions so I told them to go ahead and export `HttpFetchError` https://github.com/elastic/kibana/pull/47635 so depending on when that merges you might have to resolve that but exporting this as an interface is much better.